packages/apidom-ls/src/services/validation/linter-functions.ts

```diff
@@ -1219,4 +1220,75 @@ export const standardLinterfunctions: FunctionItem[] = [
       return true;
     },
   },
+  {
+    functionName: 'apilintSecurityScopeResolved',
+    function: (element: Element): boolean => {
+      const api = root(element);
+      const securityDefinitions = isObject(api) && api.get('securityDefinitions');
+      if (!securityDefinitions || !isObject(securityDefinitions)) return true;
+
+      const hasScope = (schemeName: string, scopesFromSecurity: string[]) => {
+        const oneSecurityDefinition = securityDefinitions.get(schemeName);
+        if (!oneSecurityDefinition) return true; // returning true, because when key is not found, then keys--defined rule will come into play.
+
+        const oneSecurityDefinitionScopes = oneSecurityDefinition.get('scopes');
+        if (!oneSecurityDefinitionScopes) return true; // returning true, because when scopes is not found, then scope--required rule from security scheme will come into play.
+
+        return scopesFromSecurity.every(
+          (scopeFromSecurity: string) => !!oneSecurityDefinitionScopes.get(scopeFromSecurity),
+        );
+      };
+
+      if (isObject(element)) {
+        const securityRequirement = element.toValue();
+        for (const [schemeName, scopesFromSecurity] of Object.entries(securityRequirement)) {
+          if (Array.isArray(scopesFromSecurity)) {
+            return hasScope(schemeName, scopesFromSecurity);
+          }
+        }
+      }
+
+      return true;
+    },
+  },
+  {
+    functionName: 'apilintSecuritySchemeUsed',
+    function: (element: Element): boolean => {
+      if (element && element.parent && isMember(element.parent)) {
+        const schemeName: string =
+          isStringElement(element.parent.key) && element.parent.key.toValue();
+        const api = root(element);
+        const globalSecurity: ObjectElement[] = isObject(api) && api.get('security');
+        if (globalSecurity) {
+          for (const secObj of globalSecurity) {
+            if (secObj.hasKey(schemeName)) {
+              return true;
+            }
+          }
+        }
+        const paths: ObjectElement = isObject(api) && api.get('paths');
+        return !!paths.findElements(
+          (e) => {
+            if (isObjectElement(e)) {
+              if (e.hasKey('security')) {
+                const opSecurity = e.get('security');
+                if (isArrayElement(opSecurity)) {
+                  return !!opSecurity.findElements(
+                    (securityRequirementObject) =>
+                      isObjectElement(securityRequirementObject) &&
```

Comment:
I think `isObject` and `isObjectElement` are equivalent, so we don't need to import both of them. 

Similarly in other places we have e.g. `isArray` and `isArrayElement` both, in existing functions as well. We could probably refactor them in the future to use only one of them, if they are truly equivalent.

```suggestion
                      isObject(securityRequirementObject) &&
```